mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -113,20 +113,14 @@ public boolean containsAttributes(Map<String, String> requiredAttributes) {
     }
 
     @JsonIgnore
-    public Optional<ContainerSkuID> getTaskExecutorContainerDefinitionId() {
+    public ContainerSkuID getTaskExecutorContainerDefinitionId() {
```

Comment:
what happens if this is not present? Also, how would this work when we have on-demand TEs?

Reply:
TEs won't be able to register anymore without skuId as part of the attributes. Is that a problem? 

how would this work when we have on-demand TEs? I guess it won't work for now. Are we interested in making things work for on-demand as well?

Reply:
@fdc-ntflx can you confirm (maybe add to the PR context) about the decision of whether or not going fwd we still want to support any on-demand/bring-your-own-container kind of scenarios?

Reply:
I've been thinking about this a lot and decided to leave the semantics as is today. This is the reasoning. 

For the scheduler to be capable of making informed scheduling decisions, it is crucial that TEs are registered in the system with both the machine definition and scheduling constraints. Two possible strategies have been evaluated:

1. Requiring TEs to provide the SKU-ID they are associated with at the time of registration
2. Permitting TEs to communicate their scheduling constraints during registration

Each of these options carries its own set of benefits and drawbacks. The first strategy simplifies the system through a SKU-ID-centric approach, yet it also represents a major constraint, limiting TEs to register only when a predefined resource cluster is present. This restriction leads to significant modifications in mantis-helm as well and necessitates the deployment of a specialized resource cluster provider in the oss. 

The second strategy, though it may not entirely resolve the ambiguity regarding SKU-ID usage, does have the advantage of preserving existing system characteristics. This includes detaching TE registration from resource cluster definition, enabling TEs to register whenever needed, and scheduling workers "on-demand". These features eliminate the need for additional modifications in mantis-helm or introducing new providers in the oss. However, a downside of this approach is that it mandates a redeployment of agents each time new scheduling constraints are introduced or existing ones need tweaking, ensuring they match the revised scheduling constraints.

Option 2 is the one we're going with.

Reply:
@Andyz26 I also updated the internal design doc with the same section. Let me know if that's clear now.